docs/HowToFAQ/ERROR_InsufficientReads.md

```diff
@@ -0,0 +1,48 @@
+---
+title: "Error: insufficient reads clause to invoke function"
+---
+
+Example: This code
+```
+{% include_relative ERROR_InsufficientReads.dfy %}
+```
+produces this output:
+```
+{% include_relative ERROR_InsufficientReads.txt %}
+```
+
+This error message indicates that a nested call of a function needs a bigger `reads` set than its enclosing function provides.
```

Comment:
Why is it insufficient? Does `A.Valid` not have an empty reads clause? Doesn't `reads things` indicate that all members of elements in things can be read? How would you make the reads clause sufficient?

Edit: a I see that `{:autocontracts}` is what causes it to fail. Is this a good example to help with understanding reads clauses?

Reply:
I left the autocontract because it was part of the original question. The remaining problem was that Valid() needs `reads this, things`, not just `reads things`